orders-redis/src/main/java/io/helidon/examples/sockshop/orders/redis/RedisOrderRepository.java

```diff
@@ -15,11 +20,44 @@
  * that that uses Redis (via Redisson) as a backend data store.
  */
 @ApplicationScoped
-@Specializes
+@Alternative
+@Priority(Interceptor.Priority.APPLICATION+10)
 @Traced
-public class RedisOrderRepository extends DefaultOrderRepository {
```

Comment:
The sync one could actually still extend `DefaultOrderRepository`, as it simply changes the underlying storage map to use.

I would split async implementation into `OrderRepositoryAsync/RedisOrderRepositoryAsync`, per example in `Carts` service, and move that change to a separate PR so we don't conflate things.

Reply:
As discussed, I'll do this in this ticket, but we need to revisit how `RedisOrderRepository` is implemented.